app/vmalert/rule/utils.go

```diff
@@ -105,3 +105,11 @@ func isSecreteHeader(str string) bool {
 	}
 	return false
 }
+
+func isPartialResponse(res datasource.Result) bool {
+	var v bool
+	if res.IsPartial != nil && *res.IsPartial {
```

Comment:
I think there might be a small issue here. You might have meant `if res != nil && *res.IsPartial { ... }` instead? 

The current version could potentially panic if `res` is nil. What do you think?